src/zkbob/ZkBobPool.sol

```diff
@@ -302,28 +303,31 @@ abstract contract ZkBobPool is IZkBobPool, EIP1967Admin, Ownable, Parameters, Zk
         (uint256 total, uint256 totalFee, uint256 hashsum, bytes memory message) =
             direct_deposit_queue.collect(_indices, _out_commit);
 
-        (,, uint256 txCount) = _recordOperation(address(0), int256(total));
-        uint256 _pool_index = txCount << 7;
+        (IZkBobAccounting acc, uint96 poolIndex) = (accounting, pool_index);
+        if (address(acc) != address(0)) {
+            acc.recordOperation(IZkBobAccounting.TxType.AppendDirectDeposits, address(0), int256(total));
```

Comment:
it looks like TVL will be updated with a value that does not include fees whereas `transfer_token_delta` in `transact` includes fees:
https://github.com/zkBob/zkbob-contracts/blob/0d46f96fad54f5c6b55b59ee11aebc86cd654a9a/src/zkbob/ZkBobPool.sol#L215